src/app/clusters/content-launch-server/content-launch-server.cpp

```diff
@@ -39,32 +39,159 @@
+        return true;
+    }
+    return false;
+}
+} // namespace
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace ContentLauncher {
+
+void SetDelegate(EndpointId endpoint, Delegate * delegate)
+{
+    uint16_t ep = emberAfFindClusterServerEndpointIndex(endpoint, chip::app::Clusters::ContentLauncher::Id);
+    if (ep != 0xFFFF)
+    {
+        gDelegateTable[ep] = delegate;
+    }
+    else
+    {
+    }
+}
+
+} // namespace ContentLauncher
+} // namespace Clusters
+} // namespace app
+} // namespace chip
+
+// -----------------------------------------------------------------------------
+// Attribute Accessor Implementation
+
+namespace {
+
+class ContentLauncherAttrAccess : public app::AttributeAccessInterface
+{
+public:
+    ContentLauncherAttrAccess() :
+        app::AttributeAccessInterface(Optional<EndpointId>::Missing(), chip::app::Clusters::ContentLauncher::Id)
+    {}
+
+    CHIP_ERROR Read(const app::ConcreteReadAttributePath & aPath, app::AttributeValueEncoder & aEncoder) override;
 
-ContentLaunchResponse contentLauncherClusterLaunchContent(chip::EndpointId endpointId,
-                                                          std::list<ContentLaunchParamater> parameterList, bool autoplay,
-                                                          const chip::CharSpan & data);
+private:
+    CHIP_ERROR ReadAcceptsHeaderAttribute(app::AttributeValueEncoder & aEncoder, Delegate * delegate);
+    CHIP_ERROR ReadSupportedStreamingProtocols(app::AttributeValueEncoder & aEncoder, Delegate * delegate);
+};
 
-ContentLaunchResponse contentLauncherClusterLaunchUrl(const chip::CharSpan & contentUrl, const chip::CharSpan & displayString,
-                                                      ContentLaunchBrandingInformation & brandingInformation);
+ContentLauncherAttrAccess gContentLauncherAttrAccess;
+
+CHIP_ERROR ContentLauncherAttrAccess::Read(const app::ConcreteReadAttributePath & aPath, app::AttributeValueEncoder & aEncoder)
+{
+    EndpointId endpoint = aPath.mEndpointId;
+    Delegate * delegate = GetDelegate(endpoint);
+    if (SendStatusIfDelegateNull(endpoint))
+    {
+        return CHIP_NO_ERROR;
+    }
+
+    switch (aPath.mAttributeId)
+    {
+    case app::Clusters::ContentLauncher::Attributes::AcceptsHeaderList::Id: {
+        return ReadAcceptsHeaderAttribute(aEncoder, delegate);
+    }
+    case app::Clusters::ContentLauncher::Attributes::SupportedStreamingProtocols::Id: {
+        return ReadSupportedStreamingProtocols(aEncoder, delegate);
+    }
+    default: {
+        break;
+    }
+    }
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR ContentLauncherAttrAccess::ReadAcceptsHeaderAttribute(app::AttributeValueEncoder & aEncoder, Delegate * delegate)
+{
+    std::list<std::string> acceptsHeaderList = delegate->HandleGetAcceptsHeaderList();
+    return aEncoder.EncodeList([acceptsHeaderList](const auto & encoder) -> CHIP_ERROR {
+        for (std::string acceptedHeader : acceptsHeaderList)
```

Comment:
This is making copies of all the strings.  Why?  Why not `const auto & acceptedHeader : acceptsHeaderList`?

Reply:
Updated.